src/App.jsx

```diff
@@ -1,58 +1,64 @@
 import 'bulma/css/bulma.css';
 import './App.scss';
+import cn from 'classnames';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
-
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+import { useState } from 'react';
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { prepareListOfGoods } from './services/prepareListOfGoods';
+import {
+  goodsFromServer, SORT_METHOD_ALPHABET, SORT_METHOD_LENGTH,
+} from './variables/variables';
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods
+    = prepareListOfGoods(goodsFromServer, sortField, reversed);
+
+  const reset = () => {
+    setSortField('');
+    setReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_METHOD_ALPHABET)}
+          type="button"
+          className={`button is-info ${cn({ 'is-light': sortField !== SORT_METHOD_ALPHABET })}`}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => setSortField(SORT_METHOD_LENGTH)}
+          type="button"
+          className={`button is-success ${cn({ 'is-light': sortField !== SORT_METHOD_LENGTH })}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReversed(!reversed)}
```

Comment:
You can write a function handler for it, not use anonim function ( not critical)